File: src/game/powerups/PistolPowerUp.java
package game.powerups;

import game.Bomberman;
import game.entity.models.BomberEntity;
import game.entity.models.Coordinates;
import game.items.PistolItem;
import game.utils.Paths;

import java.awt.image.BufferedImage;

public class PistolPowerUp extends PowerUp {
    /**
     * Constructs a PowerUp entity with the specified coordinates.
     *
     * @param coordinates the coordinates of the PowerUp entity
     */
    public PistolPowerUp(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public BufferedImage getImage() {
        return loadAndSetImage(Paths.INSTANCE.getPowerUpsFolder() + "/pistol.png");
    }

    @Override
    public int getDuration() {
        return 30;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        Bomberman.getMatch().give(entity, new PistolItem());
    }

    @Override
    protected void cancel(BomberEntity entity) {
        Bomberman.getMatch().removeItem(entity);
    }

    @Override
    public boolean isDisplayable() {
        return false;
    }
}

File: src/game/powerups/PistolPowerUp.kt
package game.powerups

import game.Bomberman
import game.entity.EntityTypes
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.items.PistolItem
import game.utils.Paths.itemsPath
import java.awt.image.BufferedImage

class PistolPowerUp(coordinates: Coordinates?) : PowerUp(coordinates) {
    constructor(id: Long) : this(null) {
        this.id = id
    }

    override fun getImage(): BufferedImage =
            loadAndSetImage("$itemsPath/pistol.png")

    override val duration: Int
        get() = 30

    override fun doApply(entity: BomberEntity) {
        Bomberman.getMatch().give(entity, PistolItem(),true)
    }

    override fun cancel(entity: BomberEntity) {
        Bomberman.getMatch().removeItem(entity)
    }

    override val isDisplayable: Boolean
        get() = false

    override fun canPickUp(entity: BomberEntity): Boolean = true

    override fun getType(): EntityTypes = EntityTypes.PistolPowerUp
}
